Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -253,6 +212,65 @@ - (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
     }
 }
 
+- (void)addTraceWithoutTransactionToTask:(NSURLSessionTask *)sessionTask
+{
+    SentryPropagationContext *propagationContext = SentrySDK.currentHub.scope.propagationContext;
+    [self addBaggageHeader:[[propagationContext traceContext] toBaggage]
+               traceHeader:[propagationContext traceHeader]
+                 toRequest:sessionTask];
+}
+
+- (void)addBaggageHeader:(SentryBaggage *)baggage
+             traceHeader:(SentryTraceHeader *)traceHeader
+               toRequest:(NSURLSessionTask *)sessionTask
+{
+    if ([self sessionTaskRequiresPropagation:sessionTask]) {
```

Comment:
Would prefer an early return on the inverse condition to bump the much larger piece of code below out one scope indent level:
```suggestion
    if (![self sessionTaskRequiresPropagation:sessionTask]) {
        SENTRY_LOG_DEBUG(@"Not adding trace_id and baggage headers for %@",
             sessionTask.currentRequest.URL.absoluteString);
        return;
    }
    
    NSString *baggageHeader = @"";
    ...
```